core-war/src/main/webapp/jobStartPagePeas/jsp/javascript/vuejs/admin-navigation.js

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (C) 2000 - 2021 Silverpeas
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * As a special exception to the terms and conditions of version 3.0 of
+ * the GPL, you may redistribute this Program in connection with Free/Libre
+ * Open Source Software ("FLOSS") applications as described in Silverpeas's
+ * FLOSS exception.  You should have received a copy of the text describing
+ * the FLOSS exception, and it is also available here:
+ * "https://www.silverpeas.org/legal/floss_exception.html"
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+//# sourceURL=/jobStartPagePeas/jsp/javascript/vuejs/admin-navigation.js
+
+(function() {
+
+  sp.i18n.load('org.silverpeas.mylinks.multilang.myLinksBundle');
+
+  const adminNavigationAsyncComponentRepository = new VueJsAsyncComponentTemplateRepository(
+      webContext + '/jobStartPagePeas/jsp/javascript/vuejs/admin-navigation-templates.jsp');
+
+  Vue.component('admin-navigation',
+      adminNavigationAsyncComponentRepository.get('root', {
+        mixins : [VuejsApiMixin],
+        data : function() {
+          return {
+            selectorSpaces : undefined,
+            selectorCurrentSpace : undefined,
+            tree : undefined
+          };
+        },
+        created : function() {
+          this.extendApiWith({
+            injectData : function(data) {
+              data = extendsObject({
+                selectorSpaces : undefined,
+                selectorCurrentSpace : undefined,
+                path : undefined,
+                spaces : undefined,
+                applications : undefined
+              }, data);
+              this.selectorSpaces = data.selectorSpaces;
+              this.selectorCurrentSpace = data.selectorCurrentSpace;
+              const currentTreeLevel = [];
+              currentTreeLevel.path = data.path;
+              Array.prototype.push.apply(currentTreeLevel, data.spaces);
```

Comment:
Why don't call directly `currentTreeLevel.push(...)` method as we now here `currentTreeLevel` is well an array?

Reply:
After some inconveniences in the past, I took the habit to do that way.